lizmap/modules/lizmap/classes/lizmapProject.class.php

```diff
@@ -912,6 +912,79 @@ public function hasLoginFilteredLayers()
         return false;
     }
 
+    public function getLoginFilteredConfig($layername)
+    {
+        if (!$this->hasLoginFilteredLayers()) {
+            return Null;
+        }
+
+        $ln = $layername;
+        // In case $layername is a WFS TypeName
+        $layerByTypeName = $this->findLayerByTypeName($layername);
+        if($layerByTypeName){
+            $ln = $layerByTypeName->name;
+        }
+
+        if (!property_exists($this->cfg->loginFilteredLayers, $ln)) {
+            return Null;
+        }
+
+        return $pConfig->loginFilteredLayers->{$n};
+    }
+
+    public function getLoginFilters($layers)
+    {
+        $filters = array();
+
+        if (!$this->hasLoginFilteredLayers()) {
+            return $filters;
+        }
+
+        foreach ($layers as $layername) {
+            $lname = $layername;
+
+            // In case $layername is a WFS TypeName
+            $layerByTypeName = $this->findLayerByTypeName($layername);
+            if($layerByTypeName){
+                $lname = $layerByTypeName->name;
+            }
+
+            // Get config
+            $loginFilteredConfig = $this->getLoginFilteredConfig($lname);
+            if ($loginFilteredConfig == Null) {
+                continue;
+            }
+
+            // attribute to filter
+            $attribute = strtolower($loginFilteredConfig->filterAttribute);
+
+            // default no user connected
+            $filter = "\"${attribute}\" = 'all'";
```

Comment:
I guess that `$filter` content is used into an SQL query. If true, I think we should escape the content of the $attribute, to avoid SQL injection.

Similar remark in other places in the patch.

Reply:
`$filter` is not for sql, it is an URL param for QGIS

Reply:
which is used by Qgis to do an SQL query, right?

Reply:
not directly, the filter is parsed by QGIS before sending it to the datasource